lib/internal/modules/cjs/loader.js

```diff
@@ -84,6 +85,25 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  const stack = err.stack.split('\n');
+
+  const lineWithErr = stack[1];
+
+  /*
+    The regular expression below targets the most common import statement
+    usage. However, some cases are not matching, cases like import statement
+    after a comment block and/or after a variable definition.
+  */
+  if (err.message.startsWith('Unexpected token export') ||
+    (/^\s*import(?=[ {'"*])\s*(?![ (])/).test(lineWithErr)) {
+    console.warn('To load an ES module, set "type": "module" ' +
```

Comment:
I think we'd want to use `process.emitWarning()` instead to allow the end user to decide how to handle warnings?

Reply:
I can't seem to get the tests passing I changed to listen for `.on('warning',(warning) => ..assertions..)` or `.on('close',...)` also debugged the stderr output but nothing is showing up,  the arguments `--trace-warnings` is provided to the spawned process, but no warning is emitted. In that line, I just replaced with the console.warn with 

```
 process.emitWarning('To load an ES module, set "type": "module" ' +
      'in the package.json or use the .mjs ' +
      'extension.\n\n');
      'extension.\n\n', 'ExperimentalWarning', undefined);
```

Reply:
I don't think we want it as an `ExperimentalWarning` because we'll want it even after ES modules are no longer experimental. I'd say stick with the default (which is just `Warning`):

```js
process.emitWarning('To load an ES module, set "type": "module" ' +
      'in the package.json or use the .mjs ' +
      'extension.');
```

Reply:
I wonder if the problem you're having is that the error is thrown before the warning is emitted....

Reply:
To get it to work, you have to use the undocumented `now` argument for `process.emitWarning()`. I'll add a commit to this branch in a little bit....